sentry-ruby/lib/sentry/interfaces/single_exception.rb

```diff
@@ -29,17 +29,19 @@ def initialize(exception:, stacktrace: nil)
       @module = exception.class.to_s.split('::')[0...-1].join('::')
       @thread_id = Thread.current.object_id
       @stacktrace = stacktrace
+      @mechanism = mechanism || Mechanism.new
```

Comment:
Can we move this default assignment upstream? I think if we assign a default in `Hub#capture_exception` as well, then we can just assume `mechanism` will always be present when it reached client and interfaces?

Reply:
I moved it to `Client#event_from_exception` since we'll have to break the signature of that method otherwise since it uses positional and not keyword arguments.